### PR TITLE
Retreive meta.json from stacks for versioned objects

### LINKED ISSUE #1083: When versioned layout, get meta.json from stacks
Currently, meta.json is written to purl file system for all objects. In addition, it is also written to stacks file system for versioned objects (<object directory>/versions/meta.json). In the future, meta.json will not be written to the purl file system.

When reading meta.json, if the object is versioned, get it from the stacks file system; otherwise, get it from the purl file system.

Here is existing code for determining if versioned: https://github.com/sul-dlss/purl/blob/088fb87ac17d0af6aa1f235670e068ac241b1aae/app/services/resource_retriever.rb#L69-L71

### PR DESCRIPTION
**Pull request by aaron-collier**
Fixes #1083 

When an object is versioned, retrieve meta.json from the stacks path. Corresponding shared_configs PRs:

https://github.com/sul-dlss/shared_configs/pull/2228 (stage)
https://github.com/sul-dlss/shared_configs/pull/2229 (uat)
https://github.com/sul-dlss/shared_configs/pull/2230 (prod)

### CODE REVIEW

File: config/settings/test.yml [hunk 1]
```diff
@@ -6,6 +6,7 @@ purl_resource:
   versioned:
     public_xml: "%{root_path}/%{druid_tree}/%{druid}/versions/public.%{version_id}.xml"
     cocina: "%{root_path}/%{druid_tree}/%{druid}/versions/cocina.%{version_id}.json"
+    meta: "%{root_path}/%{druid_tree}/meta.json"
```
Comment (mjgiarlo) [inline config/settings/test.yml:9]: Can you also add a line like this to the README? There's a section that documents how to do local dev.

Comment (mjgiarlo) [inline config/settings/test.yml:9]: I'm not sure how much it matters here, but the shared_configs changes expect `meta.json` in a `%{druid_tree}/%{druid}/versions/meta.json`. Should we reflect that structure in the test config so it's more similar, like we do with public XML and cocina?

Comment (aaron-collier) [reply config/settings/test.yml:9]: Good catch, I meant to update that, thanks.

File: config/settings.yml [hunk 2]
```diff
@@ -24,6 +24,7 @@ purl_resource:
   versioned:
     public_xml: "https://purl.stanford.edu/%{druid}/version/%{version_id}.xml"
     cocina: "https://purl.stanford.edu/%{druid}/version/%{version_id}.json"
+    meta: "https://purl.stanford.edu/%{druid}.meta_json" # versioned meta is not supported
```
Comment (mjgiarlo) [inline config/settings.yml:27]: Should the `.` after the druid be a `/`?

Comment (aaron-collier) [reply config/settings.yml:27]: Ah, yes, but apparently not used... Blah. Fixing.

**Review by mjgiarlo — APPROVED** [review]

**Pull request merged**